src/topp/DatabaseSuitability.cpp

```diff
@@ -0,0 +1,367 @@
+      bool is_novo = true;
+      for (const String& acc : accessions)
+      {
+        if (acc.find(Constants::UserParam::CONCAT_PEPTIDE) == String::npos)
+        {
+          is_novo = false;
+          break;
+        }
+      }
+
+      if (is_novo) // top hit is de novo hit
+      {
+        if (hits.size() == 1)
+        {
+          ++count_novo;
+          continue;
+        }
+
+        PeptideHit second_hit = hits[1];
+
+        if (second_hit.getMetaValue("target_decoy") == "decoy")
+        {
+          ++count_novo;
+          continue;
+        }
+
+        // check if second hit is db hit
+        set<String> second_accessions = top_hit.extractProteinAccessionsSet();
+        bool is_novo_too = true;
+        for (const String& acc2 : second_accessions)
+        {
+          if (acc2.find(Constants::UserParam::CONCAT_PEPTIDE) == String::npos)
+          {
+            is_novo_too = false;
+            break;
+          }
+        }
+
+        if (is_novo_too) // second hit is also de novo hit
+        {
+          ++count_novo;
+        }
+        else // second hit is db hit
+        {
+          ++count_interest;
+          // check for re-ranking
+          if (no_re_rank)
+          {
+            ++count_novo;
+            continue;
+          }
+
+          if (double(top_hit.getMetaValue("MS:1002252")) - double(second_hit.getMetaValue("MS:1002252")) <= cut_off)
+          {
+            ++count_db;
+            ++count_re_ranked;
+          }
+          else ++count_novo;
+        }
+      }
+      else ++count_db; // top hit is db hit
+    }
+
+    // spectra quality
+
+    UInt64 count_novo_seq = 0;
+    set<AASequence> unique_novo;
+
+    for (const auto& pep_id : novo_peps)
+    {
+      if (pep_id.getHits().empty()) continue;
+      ++count_novo_seq;
+      unique_novo.insert(pep_id.getHits()[0].getSequence());
+    }
+
+    //-------------------------------------------------------------
+    // writing output
+    //-------------------------------------------------------------
+
+    double quality = double(count_db) / (count_db + count_novo);
+    double id_rate = double(count_novo_seq) / count_ms2_lvl;
+
+    OPENMS_LOG_INFO << count_db << " top hits that were found in the database." << endl;
+    OPENMS_LOG_INFO << count_novo << " top hits that were only found in the concatenated de novo peptide." << endl;
+    OPENMS_LOG_INFO << count_interest << " times scored a de novo hit just above a database hit. Of those times " << count_re_ranked << " top de novo hits where re-ranked using a decoy cut-off of " << cut_off << endl;
+    OPENMS_LOG_INFO << "Database suitability: " << quality << endl << endl;
+    OPENMS_LOG_INFO << unique_novo.size() << " unique de novo sequences" << endl;
+    OPENMS_LOG_INFO << count_novo_seq << " total de novo sequences" << endl; 
+    OPENMS_LOG_INFO << count_ms2_lvl << " ms2 spectra." << endl;
+    OPENMS_LOG_INFO << "MS2 ID rate: " << id_rate << endl << endl;
```

Comment:
not the classical way to interpret ID rate. Mention that this only counts deNovo IDs